dev/AppNotifications/AppNotificationActivatedEventArgs.cpp

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "AppNotificationActivatedEventArgs.h"
+#include "Microsoft.Windows.AppNotifications.AppNotificationActivatedEventArgs.g.cpp"
+#include "AppNotificationBuilder/AppNotificationBuilderUtility.h"
+
+namespace winrt::Microsoft::Windows::AppNotifications::implementation
+{
+    winrt::Windows::Foundation::Collections::IMap<hstring, hstring> AppNotificationActivatedEventArgs::DecodeArguments(std::wstring arguments)
+    {
+        auto result{ winrt::single_threaded_map<winrt::hstring, winrt::hstring>() };
+
+        std::vector<std::wstring> pairs{};
+        size_t pos{ 0 };
+
+        // Separate the key/value pairs by ';' as the delimiter
+        while ((pos = arguments.find(L';')) != std::wstring::npos)
+        {
+            pairs.push_back(arguments.substr(0, pos));
+            arguments.erase(0, pos + 1);
+        }
+
+        // Need to push back final string
+        pairs.push_back(arguments);
+
+        for (auto pair : pairs)
+        {
+            // Get the key/value individual values separated by '='
+            pos = pair.find(L'=');
+            if (pos == std::wstring::npos)
+            {
+                result.Insert(Decode(pair).c_str(), L"");
```

Comment:
So is it expected to return empty string as the value for a pair? Can we get away with not putting in anything here?

Reply:
Result is a pair of <hstring, hstring. As fas I know there is anything else we can put in this to indicate no value. insert() insists on an hstring won't accept a nullptr.

And since this is part of the IDL, we can really change it to an optional<>

Reply:
I don't think you can use optional in midl 3, but I think it's fine to have an empty "" value. Developers doing AddArgument(L"Key", L""), is valid. The toolkit allows this behavior as well. In the future, we can add AddArgument(key) for a more explicit "I want an empty value".